commons/src/main/java/io/contek/invoker/commons/websocket/BaseWebSocketApi.java

```diff
@@ -92,6 +92,9 @@ private void afterDisconnect() {
       sessionHolder.set(null);
       synchronized (components) {
         components.afterDisconnect();
+        synchronized (authenticator) {
```

Comment:
Nice catch!

Reply:
Should lock authenticator.onMessage(message); in forward message as well.
Also, in both places, please lock components and authenticator in the same order to avoid deadlock